eth/db/storage.py

```diff
@@ -151,7 +78,7 @@ def _get_write_trie(self) -> HexaryTrie:
         return self._write_trie
 
     def _get_read_trie(self) -> HexaryTrie:
-        storage_root = self._account_db.get_storage_root(self._address)
+        storage_root = self._storage_root
```

Comment:
I *think* that this first conditional means that if we start with a state_root of `BLANK_ROOT_HASH` then we'll never read our own writes. Though, that must be wrong, I'm surprised that none of the state tests would exercise that!

Reply:
Hm, not sure why that wouldn't be exercised. I guess it's possible. The test would have to run on a pre-Byzantium chain, and modify state in a second transaction (we don't ever read out storage after trie writes, since Byzantium).

Anyhow, it seems vestigial from the old handling of account deletes, and I think it's cleaner to just drop it... pushed.